crates/rpc/rpc-eth-api/src/core.rs

```diff
@@ -500,9 +500,11 @@ where
         hash: B256,
     ) -> RpcResult<Option<RpcTransaction<T::NetworkTypes>>> {
         trace!(target: "rpc::eth", ?hash, "Serving eth_getTransactionByHash");
-        Ok(EthTransactions::transaction_by_hash(self, hash).await?.map(|tx| {
-            tx.into_transaction(self.tx_resp_builder()).expect("failed to fill transaction")
-        }))
+        Ok(EthTransactions::transaction_by_hash(self, hash)
+            .await?
+            .map(|tx| tx.into_transaction(self.tx_resp_builder()))
+            .transpose()
+            .map_err(|err| internal_rpc_err(format!("failed to convert transaction: {}", err)))?)
```

Comment:
```suggestion
            .map(|tx| tx.into_transaction(self.tx_resp_builder()))
            .transpose()?
```
doesn't this work?

Reply:
I couldn't get this to work. no matter what I try its not able to convert the error types.

Reply:
I'll try again later to see if I can fix that.

Reply:
hm ok, perhaps you must explicitly add the trait bound on the trait impl in that case, smthg like
```rust
impl<T>
    EthApiServer<
        RpcTransaction<T::NetworkTypes>,
        RpcBlock<T::NetworkTypes>,
        RpcReceipt<T::NetworkTypes>,
    > for T
where
    T: FullEthApi,
    jsonrpsee_types::error::ErrorObject<'static>: From<T::Error>,
    jsonrpsee_types::error::ErrorObject<'static>: From<<T::TransactionCompat as TransactionCompat>::TransactionError>,
```